llm_studio/src/losses/text_causal_classification_modeling_losses.py

```diff
@@ -0,0 +1,53 @@
+import logging
+from typing import Any, KeysView
+
+from torch import nn
+
+__all__ = ["Losses"]
+
+
+logger = logging.getLogger(__name__)
+
+
+class CrossEntropyLoss(nn.Module):
+    def __init__(self, cfg: Any):
+        super().__init__()
+        self.cfg = cfg
+        self.loss_fn = nn.CrossEntropyLoss()
+
+    def forward(self, logits, labels):
```

Comment:
For sake of code simplicity, I think it's sufficient for now to let the user specify Num Classes manually.

I'd add some exception to the loss, however to inform the user if he/she forgets about this (like I did :) ). Something like
"labels contain [7,5,4], you only selected Num Classes = 1. Please increase Num Classes to account for all classes in the dataset." Otherwise the error reads `RuntimeError: CUDA error: device-side assert triggered` which may confuse the user. This check could also go into the dataset class.

Reply:
fixed by dataset assertions